### PR TITLE
For #18272 - [Edit card] Display a saved card information in a Edit card screen

### LINKED ISSUE #18272: [Edit card] Display a saved card information in a Edit card screen
When an user clicks on a saved click, it should display a "Edit card" screen. We should try to reuse the credit card editor layout and fragment if possible,  and fill out the appropriate information from the selected saved card in the form.

Depends on https://github.com/mozilla-mobile/fenix/issues/18269

This issue only handles displaying the information and does not include updating the credit card.

Screen Shot 2021-03-02 at 3 51 48 PM

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #18272 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by Mugurell — COMMENTED** [review]
Looks good but can we add more tests?
Especially around the `setup..` methods of the `CreditCardEditorFragment`. I feel like they are important to protect against regressions.

Also regarding the `isEditing` property and then the argument null checks: if other UX differences are expected it might be worth to have two fragments sharing common code from maybe an abstract class and so keep each fragment tied to a particular usecase: cleaner code, easier to test, easy to read and refactor later. Just a thought.

**Comment by gabrielluong** [comment]
> Looks good but can we add more tests?
> Especially around the `setup..` methods of the `CreditCardEditorFragment`. I feel like they are important to protect against regressions.
> 

Sure will do

> Also regarding the `isEditing` property and then the argument null checks: if other UX differences are expected it might be worth to have two fragments sharing common code from maybe an abstract class and so keep each fragment tied to a particular usecase: cleaner code, easier to test, easy to read and refactor later. Just a thought.

I am not expecting more states. There is a commonality because you would expect that the credit card validation will be saved between saving and updating. I think this is a good point to keep thinking about. Definitely worth revisiting and making a judgement call in the future.

**Comment by gabrielluong** [comment]
@Mugurell should be ready for review again. The refactor to include the CreditCardEditorView yielded excellent results from a testing point of view.

**Review by Mugurell — APPROVED** [review]
image

File: app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    private fun saveCreditCard() {
+        containerView.hideKeyboard()
+
+        interactor.onSaveButtonClicked(
+            UpdatableCreditCardFields(
+                billingName = name_on_card_input.text.toString(),
+                cardNumber = card_number_input.text.toString(),
+                expiryMonth = (expiry_month_drop_down.selectedItemPosition + 1).toLong(),
+                expiryYear = expiry_year_drop_down.selectedItem.toString().toLong(),
+                cardType = "amex"
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt:102]: I will do this in my immediate follow up to not block landing this while CI is green.

**Pull request merged**